src/d_main.c

```diff
@@ -272,9 +279,12 @@ void D_Display (void)
     }
 
   // draw the view directly
-  if (gamestate == GS_LEVEL && gametic)
+  if (gamestate == GS_LEVEL && (!automapactive || automapoverlay) && gametic)
```

Comment:
... this is the check in question. Specifically, `!automapactive` was replaced with `(!automapactive || automapoverlay)`. I understand if this is still too intrusive.

Reply:
That's alright.